src/js/utils/FormUtil.js

```diff
@@ -0,0 +1,209 @@
+import Util from './Util';
+
+function getProp(key) {
+  return key.split('[')[0];
+}
+
+function getPropIndex(key) {
+  let value = key.split('[');
+  return parseInt(value[1].split(']')[0])
+}
+
+function getPropKey(key) {
+  return key.split('.')[1];
+}
+
+function isNotMultipleProp(key) {
+  return !key.includes('[') || !key.includes(']');
+}
+
+function containsMultipleProp(prop, id, fieldColumn) {
+  return !!(
+    fieldColumn &&
+    fieldColumn.name &&
+    fieldColumn.name.includes(`${prop}[${id}]`)
+  );
+}
+
+function getNewDefinition(prop, propID, parentDefinition, itemDefinition, getRemoveButton, model) {
+  let newDefinition = FormUtil.getMultipleFieldDefinition(
+    prop,
+    propID,
+    itemDefinition,
+    model
+  );
+
+  // Add remove button.
+  if (getRemoveButton) {
+    newDefinition.push(
+      getRemoveButton(parentDefinition, prop, propID)
+    );
+  }
+
+  return newDefinition;
+}
+
+const FormUtil = {
+  /**
+   * @return {Array} parentDefinition the modified definition.
+   * @param {Object} options Object containing properties for
+   * transformingDefinition. The following properties are:
+   *
+   * {String} prop Property to transform into multiple fields.
+   * {Number} propID ID to assign to field. ex: port[propID].value
+   * {Object} parentDefinition Definition to modify.
+   * {Object} itemDefinition Definition to create instance from.
+   * {Number} instancesIndex Index to remove and add at.
+   * {Function} getRemoveButton Getter to return button at end of row.
+   * {Function} getNewRowButton Getter to return button at end of definition.
+   * {Array} startValues Values to default to.
+   */
+  transformDefinition: function (options) {
+    let {
+      prop,
+      propID,
+      parentDefinition,
+      itemDefinition,
+      instancesIndex,
+      getRemoveButton,
+      getNewRowButton,
+      startValues
+    } = options;
+
+    if (prop == null || parentDefinition == null || itemDefinition == null) {
+      throw 'Need prop, parent definition, and item definition to transform.'
+    }
+
+    if (instancesIndex == null) {
+      instancesIndex = 0;
+    }
+
+    if (propID == null) {
+      propID = Util.uniqueID(prop);
+    }
```

Comment:
You could move this `if` inside the `else` block in line 97, as you only use it there.